core/lib/dal/src/models/consensus/mod.rs

```diff
@@ -0,0 +1,372 @@
+                .map(proto::Transaction::build)
+                .collect(),
+            last_in_batch: Some(self.last_in_batch),
+        }
+    }
+}
+
+impl Payload {
+    pub fn decode(payload: &validator::Payload) -> anyhow::Result<Self> {
+        zksync_protobuf::decode(&payload.0)
+    }
+
+    pub fn encode(&self) -> validator::Payload {
+        validator::Payload(zksync_protobuf::encode(self))
+    }
+}
+
+impl ProtoRepr for proto::Transaction {
+    type Type = Transaction;
+
+    fn read(&self) -> anyhow::Result<Self::Type> {
+        let common_data = required(&self.common_data).context("common_data")?;
+        let execute = required(&self.execute).context("execute")?;
+        Ok(Self::Type {
+            common_data: match common_data {
+                proto::transaction::CommonData::L1(common_data) => {
+                    ExecuteTransactionCommon::L1(L1TxCommonData {
+                        sender: required(&common_data.sender_address)
+                            .and_then(|x| parse_h160(x))
+                            .context("common_data.sender_address")?,
+                        serial_id: required(&common_data.serial_id)
+                            .map(|x| PriorityOpId(*x))
+                            .context("common_data.serial_id")?,
+                        deadline_block: *required(&common_data.deadline_block)
+                            .context("common_data.deadline_block")?,
+                        layer_2_tip_fee: required(&common_data.layer_2_tip_fee)
+                            .and_then(|x| parse_h256(x))
+                            .map(h256_to_u256)
+                            .context("common_data.layer_2_tip_fee")?,
+                        full_fee: required(&common_data.full_fee)
+                            .and_then(|x| parse_h256(x))
+                            .map(h256_to_u256)
+                            .context("common_data.full_fee")?,
+                        max_fee_per_gas: required(&common_data.max_fee_per_gas)
+                            .and_then(|x| parse_h256(x))
+                            .map(h256_to_u256)
+                            .context("common_data.max_fee_per_gas")?,
+                        gas_limit: required(&common_data.gas_limit)
+                            .and_then(|x| parse_h256(x))
+                            .map(h256_to_u256)
+                            .context("common_data.gas_limit")?,
+                        gas_per_pubdata_limit: required(&common_data.gas_per_pubdata_limit)
+                            .and_then(|x| parse_h256(x))
+                            .map(h256_to_u256)
+                            .context("common_data.gas_per_pubdata_limit")?,
+                        op_processing_type: required(&common_data.op_processing_type)
+                            .and_then(|x| {
+                                OpProcessingType::try_from(u8::try_from(*x)?)
+                                    .map_err(|_| anyhow!("u8::try_from"))
```

Comment:
The inner `u8::try_from` is still needed because the value is `u32`.
Also, if fails, it returns the error. The `anyhow!` is just for `OpProcessingType::try_from` associated error, who's type is `()`.